controllers/controllers/services/instances/controller.go

```diff
@@ -125,37 +125,24 @@ func (r *Reconciler) ReconcileResource(ctx context.Context, cfServiceInstance *k
 
 	credentialsSecret, err = r.reconcileCredentials(ctx, credentialsSecret, cfServiceInstance)
 	if err != nil {
-		log.Error(err, "failed to reconcile credentials secret")
-		meta.SetStatusCondition(&cfServiceInstance.Status.Conditions, metav1.Condition{
-			Type:               CredentialsSecretAvailableCondition,
-			Status:             metav1.ConditionFalse,
-			Reason:             "FailedReconcilingCredentialsSecret",
-			Message:            err.Error(),
-			ObservedGeneration: cfServiceInstance.Generation,
-		})
+		readyConditionBuilder.
```

Comment:
I wonder whether returning an error that wraps the original one but providing additional details would be more "idiomatic". The `defer` statement in the beginning of the reconcile method would handle the error and set the `Ready` condition to `false` anyway